Podverse/PlayerHistoryItem.swift

```diff
@@ -0,0 +1,94 @@
+//
+//  PlayerHistoryItem.swift
+//  Podverse
+//
+//  Created by Mitchell Downey on 5/21/17.
+//  Copyright © 2017 Podverse LLC. All rights reserved.
+//
+
+import UIKit
+
+class PlayerHistoryItem: NSObject, NSCoding {
+    let podcastfeedUrl: String!
+    let podcastTitle: String?
+    let podcastImageUrl: String?
+    let episodeMediaUrl: String!
+    let episodeTitle: String?
+    let episodeImageUrl: String?
+    let episodeSummary: String?
+    let episodeDuration: NSNumber?
+    let episodePubDate: Date?
+    let startTime: NSNumber?     // If startTime and endTime = 0, then item is a clip, else it is an episode
+    let endTime: NSNumber?
+    let clipTitle: String?
+    let clipDescription: String?
+    let ownerName: String?
+    let ownerId: String?
+    let didFinishPlaying: Bool
+    let lastPlaybackPosition: NSNumber?
+    let lastUpdated: Date?
+    
+    required init(podcastfeedUrl:String, podcastTitle:String?, podcastImageUrl:String?, episodeMediaUrl:String, episodeTitle:String?, episodeImageUrl:String?, episodeSummary:String?, episodeDuration:NSNumber?, episodePubDate:Date?, startTime:NSNumber?, endTime:NSNumber?, clipTitle:String?, clipDescription:String?, ownerName:String?, ownerId:String?, didFinishPlaying:Bool, lastPlaybackPosition:NSNumber, lastUpdated:Date?) {
```

Comment:
If you set all the optional parameters to have a default value of nil (i.e. required init(podcastfeedUrl:String, podcastTitle:String? = nil ....) you don't need to include those variables in the initializer. this way you only include the required arguments and the arguments you specifically want to populate.